File: common/cazzar/mods/jukeboxreloaded/network/packets/PacketPlayRecord.java
package cazzar.mods.jukeboxreloaded.network.packets;

import java.util.logging.Level;

import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.tileentity.TileEntity;

import cazzar.mods.jukeboxreloaded.blocks.TileJukeBox;
import cazzar.mods.jukeboxreloaded.lib.util.LogHelper;

import com.google.common.io.ByteArrayDataInput;
import com.google.common.io.ByteArrayDataOutput;

import cpw.mods.fml.common.network.PacketDispatcher;
import cpw.mods.fml.relauncher.Side;

public class PacketPlayRecord extends PacketJukebox {
	int	x, y, z;
	String record;
	
	public PacketPlayRecord() {}
	
	public PacketPlayRecord(String record, int x, int y, int z) {
		this.record = record;
		this.x = x;
		this.y = y;
		this.z = z;
	}
	
	@Override
	public void execute(EntityPlayer player, Side side)
			throws ProtocolException {
		TileEntity te = player.worldObj.getBlockTileEntity(x, y, z);
		if (te instanceof TileJukeBox) {
			((TileJukeBox) te).forcePlayRecord(record);
			((TileJukeBox) te).setForcedPlaying(true);
		}
		else return;
		if (side.isServer()) {
			((TileJukeBox) te).markForUpdate();
			PacketDispatcher.sendPacketToAllPlayers(this.makePacket());
		}
	}
	
	@Override
	public void read(ByteArrayDataInput in) {
		x = in.readInt();
		y = in.readInt();
		z = in.readInt();
		record = in.readUTF();
	}
	
	@Override
	public void write(ByteArrayDataOutput out) {
		out.writeInt(x);
		out.writeInt(y);
		out.writeInt(z);
		out.writeUTF(record);
	}
}

File: src/main/scala/net/cazzar/mods/jukeboxreloaded/packets/PacketPlayRecord.scala
package net.cazzar.mods.jukeboxreloaded.packets

import net.cazzar.mods.jukeboxreloaded.blocks.tileentity.TileJukebox
import net.cazzar.corelib.network.packets.IPacket
import io.netty.buffer.ByteBuf
import cpw.mods.fml.common.network.ByteBufUtils._
import net.cazzar.corelib.util.ClientUtil.mc
import net.cazzar.corelib.util.CommonUtil.getSide
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.server.MinecraftServer

class PacketPlayRecord(var tile: TileJukebox, var record: String) extends IPacket {
    def this() = this(null, null)

    def read(in: ByteBuf) = {
        var x, y, z: Int = 0
        x = in.readInt()
        y = in.readInt()
        z = in.readInt()
        record = readUTF8String(in)

        if (getSide.isServer) {
            val player: EntityPlayer = MinecraftServer.getServer.getConfigurationManager.getPlayerForUsername(readUTF8String(in))
            player.worldObj.playRecord(record, x, y, z)
        } else {
            throw new RuntimeException("Cannot send this packet to the client")
        }
    }

    def write(out: ByteBuf) = {
        out.writeInt(tile.identifier().posX)
        out.writeInt(tile.identifier().posY)
        out.writeInt(tile.identifier().posZ)
        writeUTF8String(out, record)
        writeUTF8String(out, mc.thePlayer.func_146103_bH.getName)
    }
}
